From a4730018b6d386b5add0ce06a25a7a912a97a5fb Mon Sep 17 00:00:00 2001
From: "Philip, Avinash" <avinashphilip@ti.com>
Date: Thu, 8 Mar 2012 12:43:28 +0530
Subject: [PATCH 169/172] ARM: OMAP2+: pwm: Register PWM devices

This patch registers PWM devices on selective profiles by using new API's
provided for registering PWM devices.
1. EHRPWM.2 on profile #4
2. ECAP.0 device on profile #0, 1, 2 and 7

Also update platform data for HAPTICS with Max frequency of operation set
to 250 Hz as per Haptics specification (Haptic Motors Max RPM is 15000
RPM).

Signed-off-by: Philip, Avinash <avinashphilip@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c |   20 +++++++++++++++++---
 1 files changed, 17 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index ced0cc9..0d5fa7b 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -37,6 +37,7 @@
 #include <linux/pwm_backlight.h>
 #include <linux/input/ti_tscadc.h>
 #include <linux/reboot.h>
+#include <linux/pwm/pwm.h>
 
 /* LCD controller is similar to DA850 */
 #include <video/da8xx-fb.h>
@@ -912,11 +913,24 @@ static struct platform_device am335x_backlight = {
 	}
 };
 
+static struct pwmss_platform_data  pwm_pdata[3] = {
+	{
+		.version = PWM_VERSION_1,
+	},
+	{
+		.version = PWM_VERSION_1,
+	},
+	{
+		.version = PWM_VERSION_1,
+	},
+};
+
 static int __init ecap0_init(void)
 {
 	int status = 0;
 
 	if (backlight_enable) {
+		am33xx_register_ecap(0, &pwm_pdata[0]);
 		platform_device_register(&am335x_backlight);
 	}
 	return status;
@@ -1025,12 +1039,12 @@ static void uart2_init(int evm_id, int profile)
 }
 
 /* setup haptics */
-#define HAPTICS_MAX_FREQ (250)
-
+#define HAPTICS_MAX_FREQ 250
 static void haptics_init(int evm_id, int profile)
 {
 	setup_pin_mux(haptics_pin_mux);
-	return;
+	pwm_pdata[2].chan_attrib[1].max_freq = HAPTICS_MAX_FREQ;
+	am33xx_register_ehrpwm(2, &pwm_pdata[2]);
 }
 
 /* NAND partition information */
-- 
1.7.9.1

